### PR TITLE
entgql: added Description() for adding meta data to a field

### PR DESCRIPTION
**Pull request by Albert-Gao**

### CODE REVIEW

**Review by a8m — COMMENTED** [review]
Thanks for the contribution @Albert-Gao. 

The next thing is to add it to the [gqlparser/ast.Definition](https://github.com/vektah/gqlparser/blob/master/ast/definition.go#L24).

**Comment by Albert-Gao** [comment]
@a8m thanks for the comments, how to do this please? I checked the file, but do not know how to link the added the field to this FieldConfig.Descrption. What I am missing here please? Thanks :)

**Comment by giautm** [comment]
Hello @Albert-Gao, you need to add `def.Description` to this block of code. It's our query field on the Query Object.
https://github.com/ent/contrib/blob/a1f3cb44bdfe32a9ea7d1cea048b12f681ce6570/entgql/schema.go#L208-L231

The latest thing, don't forget to merge annotation if we have more than one on the entity:
https://github.com/ent/contrib/blob/a1f3cb44bdfe32a9ea7d1cea048b12f681ce6570/entgql/annotation.go#L463-L471

**Comment by Albert-Gao** [comment]
Thanks, @giautm  added, please check.

File: entgql/internal/todo/ent/schema/todo.go [hunk 1]
```diff
@@ -90,7 +90,7 @@ func (Todo) Edges() []ent.Edge {
 func (Todo) Annotations() []schema.Annotation {
 	return []schema.Annotation{
 		entgql.RelayConnection(),
-		entgql.QueryField(),
+		entgql.QueryField().Description("This the todo item"),
```
Comment (Albert-Gao) [inline entgql/internal/todo/ent/schema/todo.go:93]: I am not quite sure how to test the outcome of this, when running the tests locally, I have an error and do not know how to solve it.

image

File: entgql/schema.go [hunk 2]
```diff
@@ -223,8 +223,9 @@ func (e *schemaGenerator) buildTypes(g *gen.Graph, s *ast.Schema) error {
 			} else if ant.QueryField != nil {
 				name := ant.QueryField.fieldName(gqlType)
 				def := &ast.FieldDefinition{
-					Name: name,
-					Type: listNamedType(gqlType, false),
+					Name:        name,
+					Description: ant.QueryField.Description,
```
Comment (giautm) [inline entgql/schema.go:227]: Please `def.Description = ant.QueryField.Description` at line 219.

Comment (Albert-Gao) [reply entgql/schema.go:227]: Thanks, updated!

File: entgql/annotation.go [hunk 3]
```diff
@@ -70,6 +70,10 @@ type (
 	FieldConfig struct {
 		// Name is the name of the field in the Query object.
 		Name string `json:"Name,omitempty"`
+
+		// Description is the description of the field in the Query object.
```
Comment (a8m) [inline entgql/annotation.go:74]: godoc is not correct.
```suggestion
		// Description is the description of the field.
```

**Comment by Albert-Gao** [comment]
image

Why this test is failing locally? 🤣

File: entgql/annotation.go [hunk 4]
```diff
@@ -344,12 +348,18 @@ func QueryField(name ...string) queryFieldAnnotation {
...
-// Directives allow you apply directives to the field.
+// Directives allows you to apply directives to the field.
 func (a queryFieldAnnotation) Directives(directives ...Directive) queryFieldAnnotation {
 	a.QueryField.Directives = directives
 	return a
 }
 
+// Description allows you to apply a description to the field.
+func (a queryFieldAnnotation) Description(descriptionText string) queryFieldAnnotation {
+	a.QueryField.Description = descriptionText
```
Comment (giautm) [inline entgql/annotation.go:359]:
```suggestion
	a.QueryField.Description = text
```

File: entgql/annotation.go [hunk 5]
```diff
@@ -344,12 +348,18 @@ func QueryField(name ...string) queryFieldAnnotation {
...
 }
 
-// Directives allow you apply directives to the field.
+// Directives allows you to apply directives to the field.
 func (a queryFieldAnnotation) Directives(directives ...Directive) queryFieldAnnotation {
 	a.QueryField.Directives = directives
 	return a
 }
 
+// Description allows you to apply a description to the field.
```
Comment (giautm) [inline entgql/annotation.go:357]:
```suggestion
// Description allows you to set the description for the field.
```

**Pull request merged**